### ISSUE TITLE
Place meson.build to the root of the workspace folder to allow `subdir`

### ISSUE THREAD

**Issue by intractabilis** [xpack/vscode-xpack-extension-ts#18]
Meson rightfully prohibits sub-projects to be referenced via `..` or a full path. So if you place `meson.build` to a `meta` folder, you are really limited in what you can do inside your project. Meson needs this file to be in the root of the source tree. I attached a fully working template based on the template provided by the xpack extension. In contrast to the original template, it in fact compiles `add.c` as a shared library. It also has a few other goodies: (i) the default task is set to `xPack: build Debug`, so it can be launched with Ctrl+Shift+B; (ii) launch is configured to run `gdb` and to perform a build before starting debugging; (iii) `c_cpp_properties.json` is set up to work correctly in any configuration, in any subprojects of any subdirectory without `configurationProvider` property.

Since we discussed in another issue if `configurationProvider` is needed, you can turn off CMake Tools extension completely and see that IntelliSense is still working after removing `configurationProvider`, restarting VS Code and after resetting IntelliSense database. Shaded areas of inactive code will react correctly on switching configurations between Debug and Release.

[xpack.zip](https://github.com/xpack/vscode-xpack-extension-ts/files/6681687/xpack.zip)
